### ISSUE TITLE
can i make it work like SMLoadr used to ?

### ISSUE THREAD

**Issue by aniel300** [d-fi/releases#45]
is there a way to download from "downloadLinks.txt" and replicate download structure (naming, etc) from SMLoadr?

**Comment by nkeor** [comment]
Yes, to download from a file `d-fi -a links.txt`. To replicate the SMLoadr structure you could edit d-fi.config.json, section downloadLayout.
Not sure what the structure is so can't tell you more

**Comment by aniel300** [comment]
thank you for answering, i was able to use `d-fi -a links.txt`, as for the downloading structure/ naming i was able to use this 

>  "saveLayout": {
    "track": "Music/{ART_NAME}/{ALB_TITLE}/{SNG_TITLE}",
    "album": "Music/{ART_NAME}/{ALB_TITLE}/{SNG_TITLE}",
    "artist": "Music/{ART_NAME}/{ALB_TITLE}/{SNG_TITLE}",
    "playlist": "Playlist/{TITLE}/{SNG_TITLE}"

but it still doesn't match SMLoadr quite yet in other words is possible to make how it was including downloading ".lrc" ?
 
screenshots of SMLoadr folder and naming structure: https://imgur.com/a/W1IO4x8

**Comment by nkeor** [comment]
Downloading .lrc files is planned but not supported yet.

As for the layout, this should match yours:

```json
"saveLayout": {
  "track": "Music/{ART_NAME}/{ALB_TITLE} (Single)/{SNG_TITLE}",
  "album": "Music/{ART_NAME}/{ALB_TITLE} (Album)/{SNG_TITLE}",
  "artist": "Music/{ART_NAME}/{ALB_TITLE}/{SNG_TITLE}",
  "playlist": "Playlist/{TITLE}/{SNG_TITLE}"
},
```

**Comment by aniel300** [comment]
ok i will give it a try, thanks again.

**Comment by aniel300** [comment]
i went ahead and tried it but is not quite there: https://i.imgur.com/voGjoUu.png

**Comment by nkeor** [comment]
Ah so you want it to be `POSITION TITLE` instead of `POSITION - TITLE`...

This is currently not configurable, but I might be able to add it to the config after the upcoming refactor. No ETA yet but sometime next month.

**Comment by sayem314** [comment]
`TRACK_NUMBER` can be used for tracks and for playlist `TRACK_POSITION` should be used. This should work.

```json
"saveLayout": {
  "track": "Music/{ART_NAME}/{TRACK_NUMBER} - {ALB_TITLE} (Single)/{SNG_TITLE}",
  "album": "Music/{ART_NAME}/{TRACK_NUMBER} - {ALB_TITLE} (Album)/{SNG_TITLE}",
  "artist": "Music/{ART_NAME}/{TRACK_NUMBER} - {ALB_TITLE}/{SNG_TITLE}",
  "playlist": "Playlist/{TITLE}/{TRACK_POSITION} - {SNG_TITLE}"
},
```

For a list of available props check this file.
https://github.com/d-fi/d-fi-core/blob/4ff25adeaa26408788e42836f3449636af9c1d16/src/types/tracks.ts#L23-L95

**Comment by aniel300** [comment]
can we take a look here to see if we can try to replicate the file structure? if i remember correctly SMLoadr would create the nested file structure whether or not i gave it a artist, album or track link. 
https://git.telekem.net/nik/SMLoadr/commit/6e620ae5996651c1908206d1cfcc9dbf982b2a1e
https://git.telekem.net/nik/SMLoadr/commit/07b703ea03087a9145579e6e3dbfa2079c01dcc6?style=split

**Comment by sayem314** [comment]
You can configure the structure however you want. Take a look at available props [here](https://github.com/d-fi/d-fi-core/blob/44ba642931d564b6a91109e2a051e24b899dcfbf/src/types/tracks.ts#L23-L95)

**Comment by aniel300** [comment]
hey, hope u doing ok, does this tool still being developed?

**Comment by nkeor** [comment]
I'm still maintaining d-fi, yes

**Comment by aniel300** [comment]
thank you i really appreciate it

**Comment by aniel300** [comment]
it is getting stuck randomly after running for some time. is it because am running it directly from a smb share? how can i make it not get stuck ?
![WindowsTerminal_wjqtuLk3UC](https://user-images.githubusercontent.com/7571729/213797719-333a6dde-dcbf-4ba4-9769-a40dca3b02e4.png)

**Comment by aniel300** [comment]
@nkeor @sayem314 hi, hope u guys doing ok. how can i replicate this? 
![image](https://github.com/d-fi/releases/assets/7571729/34f92883-5b9a-400c-bb35-5f35222478f6)

**Comment by sayem314** [comment]
Any question, ask in https://t.me/dFiCommunity

**Comment by aniel300** [comment]
will do, thank you.
